icicle-compiler/src/Icicle/Sea/Preamble.hs

```diff
@@ -26,16 +26,37 @@ import           X.Data.FileEmbed (embedWhen)
 seaPreamble :: Doc
 seaPreamble
  = vsep
- $ fmap (uncurry seaOfExternal) files
+ ( fmap (uncurry seaOfExternal) externals <> fmap (uncurry seaOfHeader) files )
  where
   files
    = List.sortBy (compare `on` fst)
    $ $(embedWhen ((== ".h") . takeExtension) "data/sea/")
+  externals
+   = let base    = $(embedWhen ((== "anemone_base.h"))    "data/csrc/anemone")
+         mempool = $(embedWhen ((== "anemone_mempool.h")) "data/csrc/anemone")
+     in base <> mempool
 
 seaOfExternal :: FilePath -> ByteString -> Doc
 seaOfExternal path bs
  = vsep
  [ "// " <> text path
+ , text (T.unpack (T.unlines file))
+ , ""
+ ]
+ where
+  (_, file)
+   = case T.decodeUtf8' bs of
+       Left err ->
+         Savage.error $
+           "Icicle.Sea.Preamble.seaOfExternal: failed to decode: " <> path <> "\n" <>
+           show err
+       Right txt ->
+         List.partition (T.isPrefixOf "#include \"") $ T.lines txt
```

Comment:
I think we might be able to keep correct line numbers in both cases.. maybe?
```
lines = zipWith mkLine (T.lines txt) [1...]
mkLine line lineNo
 | T.isPrefixOf "#include \""...
 = "#line blah " <> show lineNo
 | otherwise
 = line
```